plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/ViewportPluginImpl.kt

```diff
@@ -0,0 +1,389 @@
+package com.mapbox.maps.plugin.viewport
+
+import android.animation.Animator
+import android.animation.AnimatorSet
+import android.animation.ValueAnimator
+import androidx.core.animation.doOnEnd
+import com.mapbox.maps.plugin.InvalidPluginConfigurationException
+import com.mapbox.maps.plugin.Plugin
+import com.mapbox.maps.plugin.animation.CameraAnimationsPlugin
+import com.mapbox.maps.plugin.animation.camera
+import com.mapbox.maps.plugin.delegates.MapDelegateProvider
+import com.mapbox.maps.plugin.locationcomponent.LocationComponentPlugin
+import com.mapbox.maps.plugin.viewport.ViewportPlugin.Companion.DEFAULT_FRAME_TRANSITION_OPT
+import com.mapbox.maps.plugin.viewport.data.MapboxViewportDataSource
+import com.mapbox.maps.plugin.viewport.data.ViewportData
+import com.mapbox.maps.plugin.viewport.data.ViewportDataSource
+import com.mapbox.maps.plugin.viewport.data.ViewportDataSourceUpdateObserver
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.Following
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.Idle
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.Overview
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.TransitionToFollowing
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraState.TransitionToOverview
+import com.mapbox.maps.plugin.viewport.state.ViewportCameraStateChangedObserver
+import com.mapbox.maps.plugin.viewport.transition.MapboxViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraStateTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransition
+import com.mapbox.maps.plugin.viewport.transition.ViewportCameraTransitionOptions
+import java.util.concurrent.CopyOnWriteArraySet
+
+/**
+ * Mapbox default implementation for Viewport plugin.
+ */
+class ViewportPluginImpl : ViewportPlugin {
```

Comment:
I assume impl should be `internal`

Reply:
It have to be public since we now use concrete class constructor instead of reflection to initialise the plugin instances.